striketracker/config_test.go

```diff
@@ -0,0 +1,26 @@
+package striketracker
+
+import (
+	"testing"
+)
+
+const (
+	ConfigTestToken = "98fg798hu7sd9hu32kj23kj23"
+	ConfigTestID    = "TestApplication"
+)
+
+func TestFunctionalApi(t *testing.T) {
+	config := Configuration{}
+	config.Option(WithDebug(true), WithApplicationID(ConfigTestID), WithAuthorizationHeaderToken(ConfigTestToken))
```

Comment:
This is actually not a bad pass I think without your implementing client.

I'll have to verify later and play around a bit. The client implementation looks like this (as I'm sure you've seen)

```
func NewClientWithOptions(opts ...Config) (*Client, error) {
	options := &Configuration{}
	for _, opt := range opts {
		opt(options)
	}
```

Which if I'm not mistaken, translates roughly to what you are doing

Reply:
I added `NewConfiguration` and updated the unit tests